### PR TITLE
Improve Faker connector docs

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Just a bunch of cosmetics and style fixes, plus clarification for the function.

## Additional context and related issues

Follow up to #23990 and #24008

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: docs/src/main/sphinx/connector/faker.md [hunk 1]
```diff
@@ -126,11 +125,13 @@ See the Datafaker's documentation for more information about
...
 and [available providers](https://www.datafaker.net/documentation/providers/).
 
-To test a generator expression, without having to recreate the table, use the
-`random_string` function from the `default` schema:
+Use the `random_string` function from the `default` schema of the `generator`
+catalog to test a generator expression:
 
 ```sql
-SELECT default.random_string('#{Name.first_name}')
+SELECT generator.default.random_string('#{Name.first_name}');
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/faker.md:132]: Can you confirm/test this for me @nineinchnick  ..

Comment (nineinchnick) [reply docs/src/main/sphinx/connector/faker.md:132]: Catalog name is set by the user, and we don't have examples in this section, that's why I omitted it

**Review by nineinchnick — APPROVED** [review]

File: docs/src/main/sphinx/connector/faker.md (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/faker.md:132]: thats why the sentence above includes the catalog name..  without it in the example it potentially doesnt work.

File: docs/src/main/sphinx/connector/faker.md [hunk 2]
```diff
@@ -257,8 +260,8 @@ INSERT INTO production.public.customers
 SELECT *
 FROM generator.default.customers
 WHERE
-  born_at BETWEEN CURRENT_DATE - INTERVAL '150' YEAR AND CURRENT_DATE
-  AND age_years BETWEEN 0 AND 150
+  born_at BETWEEN CURRENT_DATE - INTERVAL '80' YEAR AND CURRENT_DATE
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/connector/faker.md:263]: How many people do you know who are over 100 years old.. faker is about more realistic data....

[the remark this answers is not in the document]
Comment (nineinchnick) [reply docs/src/main/sphinx/connector/faker.md:263]: Realistic is subjective, I tried keeping this inclusive. This is also about testing boundaries and catching unexpected behavior. That's why I opted for a wider range.

File: docs/src/main/sphinx/connector/faker.md (hunk 1 above)
Comment (nineinchnick) [reply docs/src/main/sphinx/connector/faker.md:132]: So we have to mention again that this assumes the catalog has this specific name

Comment (mosabua) [reply docs/src/main/sphinx/connector/faker.md:132]: Lets test this next week together (unless I get around to it earlier myself...)

File: docs/src/main/sphinx/connector/faker.md [hunk 3]
```diff
@@ -20,83 +23,79 @@ faker.null-probability=0.1
...
 ```
 
-Create tables in the `default` schema, or create different schemas first.
-Reading from tables in this catalog return random data. See [](faker-usage) for
-more examples.
+Create tables in the `default` schema, or create different schemas first. Tables
+in the catalog only exist as definition and do not hold actual data. Any query
+reading from tables returns random data. As a result, repeated invocation of an
+identical query does not return identical data. See [](faker-usage) for more
+examples.
```
Comment (nineinchnick) [inline docs/src/main/sphinx/connector/faker.md:30]: The last part is being addressed in https://github.com/trinodb/trino/pull/24008

**Comment by mosabua** [comment]
@wendigo .. can I get an approval here?

File: docs/src/main/sphinx/connector/faker.md (hunk 3 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/faker.md:30]: I think we still need a wording here that details this more .. even after 24008 goes in - then we just need to explain that its random and deterministics.. but spell out what that means in practice. E.g. what if you create two identical tables..

Comment (nineinchnick) [reply docs/src/main/sphinx/connector/faker.md:30]: Agree! The data being deterministic will require explaining it depends on the schema - even with a constant seed, any changes to the schema will affect the order in which generators are called, and may result in getting a completely different data set.

**Comment by mosabua** [comment]
I have to update this now since #24008 merged .. and things are different .. but still need fix up.

File: docs/src/main/sphinx/connector/faker.md [hunk 4]
```diff
@@ -20,83 +23,83 @@ faker.null-probability=0.1
...
 ```
 
-Create tables in the `default` schema, or create different schemas first.
-Reading from tables in this catalog return random data. See [](faker-usage) for
-more examples.
+Create tables in the `default` schema, or create different schemas first. Tables
+in the catalog only exist as definition and do not hold actual data. Any query
+reading from tables returns random data. 
+
+WHAT DO WE EXPLAIN HERE NOW ?
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/faker.md:30]: @nineinchnick .. maybe we should work on this together

File: docs/src/main/sphinx/connector/faker.md [hunk 5]
```diff
@@ -126,11 +125,13 @@ See the Datafaker's documentation for more information about
 [the expression](https://www.datafaker.net/documentation/expressions/) syntax
 and [available providers](https://www.datafaker.net/documentation/providers/).
 
-To test a generator expression, without having to recreate the table, use the
-`random_string` function from the `default` schema:
+Use the `random_string` function from the `default` schema of the `generator`
+catalog to test a generator expression:
 
 ```sql
-SELECT default.random_string('#{Name.first_name}')
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/faker.md:133]: I am not sure if this actually works but in either case .. it relies on a default catalog being set and is therefore to implicit.

File: docs/src/main/sphinx/connector/faker.md [hunk 6]
```diff
@@ -20,83 +23,79 @@ faker.null-probability=0.1
 faker.default-limit=1000
 ```
 
-Create tables in the `default` schema, or create different schemas first.
-Reading from tables in this catalog return random data. See [](faker-usage) for
-more examples.
+Create tables in the `default` schema, or create different schemas first. Tables
+in the catalog only exist as definition and do not hold actual data. Any query
+reading from tables returns random, but deterministic data. As a result,
+repeated invocation of an identical query return identical data. See
```
Comment (nineinchnick) [inline docs/src/main/sphinx/connector/faker.md:29]:
```suggestion
repeated invocation of an query return identical data. See
```

**Review by nineinchnick — APPROVED** [review]

File: docs/src/main/sphinx/connector/faker.md (hunk 5 above)
Comment (nineinchnick) [reply docs/src/main/sphinx/connector/faker.md:133]: Let's only keep one example here - remove the second one starting with `USE`.

I tested this locally with:
```bash
docker run -d --name trino --rm -e CATALOG_MANAGEMENT=dynamic -p8080:8080 trino:465-SNAPSHOT-arm64
trino --execute "create catalog generator using faker; select generator.default.random_string('#{currency.code}');"
```

Which prints:
```
CREATE CATALOG
 _col0 
-------
 XBB   
(1 row)
```

Comment (mosabua) [reply docs/src/main/sphinx/connector/faker.md:133]: Done

**Review by losipiuk — APPROVED** [review]

**Comment by mosabua** [comment]
Thank you @losipiuk 

Test failure is a false alarm. Merging.

**Pull request merged**